src/controllers/connectionManager.ts

```diff
@@ -721,79 +721,86 @@ export default class ConnectionManager {
 	// create a new connection with the connectionCreds provided
 	public async connect(fileUri: string, connectionCreds: IConnectionInfo, promise?: Deferred<boolean>): Promise<boolean> {
 		const self = this;
```

Comment:
You should be able to get rid of this line and just use this since everything here is using lambdas